docs/decisions/0001-modular-frontend.rst

```diff
@@ -0,0 +1,68 @@
+# Modular frontends
+
+## Background
+
+A couple of issues have been identified with the current MFE architecure that
+we hope to resolve with this proposal. Those issues are:
+
+- Duplicated javascript and styling code across multiple MFEs that needs to be
+  loaded multiple times.
+- Large bundle size for each MFE that includes the entire codebase instead of
+  the smaller subset of features needed.
+- Use of different libraries, versions of libraries etc across MFEs further
+  increasing overall bundle size.
+- An increasing number of different MFEs to keep track of, build and update.
+- Poor customisability due to the lack of plugins, or other hooks into MFEs
+- No seamless routing between MFEs.
+
+## Proposal
+
+Our proposal boils down to the following:
+
+- Creating of an LMS mono-repo that pulls in MFEs related to the LMS into a
+  single repo, potentially starting as submodules and eventully being hosted in
+  the main repo.
+- Use of webpack module federation to provide both a mechanism to support loose
+  dependencies between MFEs, and for plugins
+- Updating frontend-platform to support a single shared global state for auth
+  etc, and an mfe-level state.
+- Update to react-route v6 or alternative that supports relative routes, so that
+  the parent MFE can provide a base route, and the MFE can handle routing inside
+  that.
+- Use of npm workspaces to have each MFE continue to be a separate package but
+  dependencies shared across all MFEs can be added to the parent package and
+  updated globally there.
```

Comment:
That sounds nice. I'm not familiar with it though so let's add some details. Can MFEs override the dependencies if they need to?

I would also reccommed using [pnpm workspaces](https://pnpm.io/workspaces) instead of plain npm as pnpm is not only much faster but also prevents loading implicit dependencies, which keeps the dependency tree much more rigorous and clean.

Reply:
Yes, MFEs can override dependencies of parent packages, I think, just by specifying their own.

pnpm's workspaces seem similar but more powerful. Will take a look.